### DISCUSSION TITLE
WooCommerce archive-product.php

### DISCUSSION THREAD

**Discussion by guylcollins** [gregsullivan/_tw#81] [Q&A]
I'm finding that the woocommerce archive-product.php file is completely ignored, both if created within the theme, but also from Woocommerce plugin files as well

I can make edits to content-single-product.php and they show, but not to archives?

Any help most welcome

**Comment by gregsullivan** [comment]
Hello—Thanks for posting this!

I took a look at the most recent WooCommerce site I worked on using _tw, and I have an `archive-product.php` file at `./theme/woocommerce/archive-product.php` that is working as expected.

Is that the location of the file in your theme? It would be worth confirming that you only have one copy of the file, and only at that location; short of seeing your theme repository, that would be my guess as a potential issue off the top of my head.

Hope that helps! Please let me know if there are any other details you think might be relevant, and I'd be happy to take a look.

**Reply by ghost** [reply]
I'm having a similar issue. I'm using WooCommerce and the product page is missing many of the usual elements including the add to cart button. Editing the template in ./theme/woocommerce/single-product.php or  ./theme/woocommerce/content-single-product.php seems to yield nothing. 

If I switch to a default theme, the product page works fine. I haven't made any changes to functions.php

**Reply by gregsullivan** [reply]
Have you done anything to remove WooCommerce elements via hooks?

From my side, I can say that I started work on a new WooCommerce theme in the last couple of weeks. I haven't reached the point of creating the `./theme/woocommerce` folder yet. The add-to-cart button (and, to my knowledge, all others) are present as expected.

As a first step, you may want to try renaming the `./theme/woocommerce` folder (e.g., `./theme/woocommerce-disabled`). If the missing elements appear, you can be confident it's an issue in one of the files in that folder.

If renaming the folder has no effect, I would guess WooCommerce elements are being removed via hooks.

Before doing any of the above, you'll want to confirm caching isn't affecting your ability to view any of your changes.

**Reply by ghost** [reply]
Thanks for the reply! Changing the folder name didn't effect it. I didn't add any hooks to remove anything and I don't see any baked in... I always develop with disable caching turned on. 

I did try generating a new clean version of _tw to see if I'm missing something within my theme. However, when I switched to the clean download of _tw, the add-to-cart is still missing.

**Reply by gregsullivan** [reply]
My guesses at this point would be:

* Plugin incompatibility. Do you have any other WooCommerce-specific plugins installed? They could be expecting a `woocommerce.php` file or WooCommerce support to be declared.
* Theme-specific settings from Customizer or elsewhere that are only being applied to your generated theme. (Though if you generated a clean version with a different name, that would mean it's not this. If the clean version had the same directory name, this is still a possibility.)

**Reply by gregsullivan** [reply]
And you're very welcome—thanks for posting your question! My responses are a bit shorter than usual because I'm getting caught up post-holiday.

**Reply by ghost** [reply]
I figured it out. I didn't have the "Shop Page" set in WooCommerce > Settings > Products > General. Once that was set, it worked.

**Reply by gregsullivan** [reply]
Awesome, very happy to hear it!
